schemas.py

```diff
@@ -26,21 +26,21 @@ def rapper(fn):
         def wrapt(event, context, *args):
             email = event[email_key]
             token = event[token_key]
+            users = util.coll('users')
 
-            tests = util.coll('users')
+            # try to find user
+            user = users.find_one({'email': email})
+            if user is None:
+                return on_failure(event, context, 'User Not found', *args)
 
-            #try to find our user
-            results = tests.find_one({"email":email})
-            if results is None or results == None or results == [] or results == ():
-                return on_failure(event, context, "User not found", *args)
+            # look for token
+            tokens = list(filter(lambda auth: auth['token'] == token, user['auth']))
+            if len(tokens) == 0 or datetime.now() > parse(tokens[0]['valid_until']):
+                return on_failure(event, context, 'Token invalid', *args)
```

Comment:
I'm reasonably sure these are equivalent...

But if you feel that this is more readable :man_shrugging:

Reply:
Looking back, I may have actually just read the date format wrong, but I feel this refactoring makes it a bit more obvious to tell what's going on.
On an interesting note, if we change all the auth toke stuff to date objects instead of strings, we may actually be able to check this with one query

Reply:
:laughing: 

That also makes things nicely deterministic (it's like a random string now). It'd actually be kind of a nice change if anybody feels up for it.